[geonetwork] 404 errors accessing geonetwork with all logins containing a dot

sometimes (havent figured when/how yet) when a user access the catalog there's a 404, from my understanding coming from the code added in georchestra for the gn<->ldap sync
```
Etat HTTP 500 – Erreur interne du serveur

Type Rapport d'exception

message 404 : [no body]

description Le serveur a rencontré une erreur interne qui l'a empêché de satisfaire la requête.

exception

org.springframework.web.client.HttpClientErrorException$NotFound: 404 : [no body]
	org.springframework.web.client.HttpClientErrorException.create([HttpClientErrorException.java:113](http://httpclienterrorexception.java:113/))
	org.springframework.web.client.DefaultResponseErrorHandler.handleError([DefaultResponseErrorHandler.java:170](http://defaultresponseerrorhandler.java:170/))
	org.springframework.web.client.DefaultResponseErrorHandler.handleError([DefaultResponseErrorHandler.java:112](http://defaultresponseerrorhandler.java:112/))
	org.springframework.web.client.ResponseErrorHandler.handleError([ResponseErrorHandler.java:63](http://responseerrorhandler.java:63/))
	org.springframework.web.client.RestTemplate.handleResponse([RestTemplate.java:782](http://resttemplate.java:782/))
	org.springframework.web.client.RestTemplate.doExecute([RestTemplate.java:740](http://resttemplate.java:740/))
	org.springframework.web.client.RestTemplate.execute([RestTemplate.java:674](http://resttemplate.java:674/))
	org.springframework.web.client.RestTemplate.getForEntity([RestTemplate.java:342](http://resttemplate.java:342/))
	[org.georchestra.security](http://org.georchestra.security/).client.console.RestClient.get([RestClient.java:40](http://restclient.java:40/))
	[org.georchestra.security](http://org.georchestra.security/).client.console.GeorchestraConsoleUsersApiImpl.findByUsername([GeorchestraConsoleUsersApiImpl.java:57](http://georchestraconsoleusersapiimpl.java:57/))
	[org.georchestra.geonetwork.security](http://org.georchestra.geonetwork.security/).integration.GeorchestraAccountsRepository.findUserByUsername([GeorchestraAccountsRepository.java:58](http://georchestraaccountsrepository.java:58/))
	[org.geonetwork.security](http://org.geonetwork.security/).external.integration.UserSynchronizer.findCanonicalUserByUsername([UserSynchronizer.java:83](http://usersynchronizer.java:83/))
	[org.geonetwork.security](http://org.geonetwork.security/).external.integration.UserSynchronizer$$FastClassBySpringCGLIB$$88a22663.invoke(<generated>)
	org.springframework.cglib.proxy.MethodProxy.invoke([MethodProxy.java:218](http://methodproxy.java:218/))
	org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept([CglibAopProxy.java:687](http://cglibaopproxy.java:687/))
	[org.geonetwork.security](http://org.geonetwork.security/).external.integration.UserSynchronizer$$EnhancerBySpringCGLIB$$66c33ff0.findCanonicalUserByUsername(<generated>)
	[org.geonetwork.security](http://org.geonetwork.security/).external.integration.AccountsReconcilingService.findCanonicalUserByUsername([AccountsReconcilingService.java:101](http://accountsreconcilingservice.java:101/))
	[org.geonetwork.security](http://org.geonetwork.security/).external.integration.AccountsReconcilingService.findUpToDateUserByUsername([AccountsReconcilingService.java:89](http://accountsreconcilingservice.java:89/))
	[org.geonetwork.security](http://org.geonetwork.security/).external.integration.AccountsReconcilingService$$FastClassBySpringCGLIB$$7d978086.invoke(<generated>)
	org.springframework.cglib.proxy.MethodProxy.invoke([MethodProxy.java:218](http://methodproxy.java:218/))
	org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept([CglibAopProxy.java:687](http://cglibaopproxy.java:687/))
	[org.geonetwork.security](http://org.geonetwork.security/).external.integration.AccountsReconcilingService$$EnhancerBySpringCGLIB$$e30f3c39.findUpToDateUserByUsername(<generated>)
	[org.georchestra.geonetwork.security](http://org.georchestra.geonetwork.security/).authentication.GeorchestraPreAuthenticationFilter.getPreAuthenticatedPrincipal([GeorchestraPreAuthenticationFilter.java:95](http://georchestrapreauthenticationfilter.java:95/))
	[org.georchestra.geonetwork.security](http://org.georchestra.geonetwork.security/).authentication.GeorchestraPreAuthenticationFilter.getPreAuthenticatedPrincipal([GeorchestraPreAuthenticationFilter.java:49](http://georchestrapreauthenticationfilter.java:49/))
	[org.springframework.security](http://org.springframework.security/).web.authentication.preauth.AbstractPreAuthenticatedProcessingFilter.doAuthenticate([AbstractPreAuthenticatedProcessingFilter.java:167](http://abstractpreauthenticatedprocessingfilter.java:167/))
	[org.springframework.security](http://org.springframework.security/).web.authentication.preauth.AbstractPreAuthenticatedProcessingFilter.doFilter([AbstractPreAuthenticatedProcessingFilter.java:121](http://abstractpreauthenticatedprocessingfilter.java:121/))
	[org.springframework.security](http://org.springframework.security/).web.FilterChainProxy$VirtualFilterChain.doFilter([FilterChainProxy.java:334](http://filterchainproxy.java:334/))
	[org.springframework.security](http://org.springframework.security/).web.authentication.logout.LogoutFilter.doFilter([LogoutFilter.java:116](http://logoutfilter.java:116/))
	[org.springframework.security](http://org.springframework.security/).web.FilterChainProxy$VirtualFilterChain.doFilter([FilterChainProxy.java:334](http://filterchainproxy.java:334/))
	[org.springframework.security](http://org.springframework.security/).web.csrf.CsrfFilter.doFilterInternal([CsrfFilter.java:117](http://csrffilter.java:117/))
	org.springframework.web.filter.OncePerRequestFilter.doFilter([OncePerRequestFilter.java:119](http://onceperrequestfilter.java:119/))
	[org.springframework.security](http://org.springframework.security/).web.FilterChainProxy$VirtualFilterChain.doFilter([FilterChainProxy.java:334](http://filterchainproxy.java:334/))
	[org.springframework.security](http://org.springframework.security/).web.context.SecurityContextPersistenceFilter.doFilter([SecurityContextPersistenceFilter.java:105](http://securitycontextpersistencefilter.java:105/))
	[org.springframework.security](http://org.springframework.security/).web.FilterChainProxy$VirtualFilterChain.doFilter([FilterChainProxy.java:334](http://filterchainproxy.java:334/))
	[org.springframework.security](http://org.springframework.security/).web.FilterChainProxy.doFilterInternal([FilterChainProxy.java:215](http://filterchainproxy.java:215/))
	[org.springframework.security](http://org.springframework.security/).web.FilterChainProxy.doFilter([FilterChainProxy.java:178](http://filterchainproxy.java:178/))
	org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate([DelegatingFilterProxy.java:358](http://delegatingfilterproxy.java:358/))
	org.springframework.web.filter.DelegatingFilterProxy.doFilter([DelegatingFilterProxy.java:271](http://delegatingfilterproxy.java:271/))
	jeeves.config.springutil.JeevesDelegatingFilterProxy.doFilter([JeevesDelegatingFilterProxy.java:104](http://jeevesdelegatingfilterproxy.java:104/))
	org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal([CharacterEncodingFilter.java:201](http://characterencodingfilter.java:201/))
	org.springframework.web.filter.OncePerRequestFilter.doFilter([OncePerRequestFilter.java:119](http://onceperrequestfilter.java:119/))
```

from my understanding at runtime, when a user connects to gn then gn queries the console for this user details ?

adding debug printfs, it seems the stripping happens in spring ? according to https://www.baeldung.com/spring-mvc-pathvariable-dot `@PathVariable("name")` does unwanted things with dots.....
```
$curl http://localhost:8280/console/internal/users/username/user.test
2022-09-26 11:12:15 SecurityApiController [ERROR] SecurityApiController.findUserByUsername(user)
2022-09-26 11:12:15 UsersApiImpl [ERROR] UsersApiImpl.findByUsername(user)
2022-09-26 11:12:15 AccountDaoImpl [ERROR] AccountDaoImpl.findByUID(user)

```

--- a/console/src/main/java/org/georchestra/console/ws/security/api/SecurityApiController.java
+++ b/console/src/main/java/org/georchestra/console/ws/security/api/SecurityApiController.java
@@ -70,7 +70,7 @@ public class SecurityApiController {
      * <p>
      * This is the server-side counterpart of {@link UsersApi#findByUsername}
      */
-    @GetMapping(value = "/users/username/{name}")
+    @GetMapping(value = "/users/username/{name:.+}")
     public ResponseEntity<GeorchestraUser> findUserByUsername(@PathVariable("name") String name) {
         return toEntityOrNotFound(users.findByUsername(name));
     }

